boilerplate/_lib/ensure.sh

```diff
@@ -22,6 +36,54 @@ golangci-lint)
         curl -sfL "${DOWNLOAD_URL}" | tar -C "${GOPATH}/bin" -zx --strip-components=1 "golangci-lint-${GOLANGCI_LINT_VERSION}-${GOOS}-amd64/golangci-lint"
     fi
     ;;
+
+operator-sdk)
+    #########################################################
+    # Ensure operator-sdk is installed at the desired version
+    # When done, ./.operator-sdk/bin/operator-sdk will be a
+    # symlink to the appropriate executable.
+    #########################################################
+    # First discover the desired version from go.mod
+    # The following properly takes `replace` directives into account.
+    wantver=$(go list -json -m github.com/operator-framework/operator-sdk | jq -r 'if .Replace != null then .Replace.Version else .Version end')
+    echo "go.mod says you want operator-sdk $wantver"
+    # Where we'll put our (binary and) symlink
+    mkdir -p .operator-sdk/bin
+    cd .operator-sdk/bin
+    # Discover existing, giving preference to one already installed in
+    # this path, because that has a higher probability of being the
+    # right one.
+    if [[ -x ./operator-sdk ]] && [[ "$(osdk_version ./operator-sdk)" == "$wantver" ]]; then
+        echo "operator-sdk $wantver already installed"
+        exit 0
+    fi
```

Comment:
In case operator-sdk version mismatches, it may be good to error (with a variable to skip the test?) to avoid confusion as the user running the `operator sdk` manually would lead to a different behaviour).

Reply:
mehh, don't fully agree. make targets should be a convenient way to run standard commands, but not the only easy one. Developers needs to be able to continue using standard tools easily.
If I need to debug and want to add a parameter to the `operator-sdk` command, I don't want to have to clone boilerplate, update it and then set the ENV variable overiding the GIT repo to finally succeed to run the `make` command. I should be able to easily run the same command (and here, version mismatch would be a pain for the user)

Reply:
Oh, absolutely. But if the developer is running their own binary, they're responsible for making sure it's the right version. There's nothing we can do to help them with that -- except for providing a convenient `make` target.